packages/langchain/src/openai/chat.ts

```diff
@@ -0,0 +1,74 @@
+import { CallbackManagerForLLMRun } from '@langchain/core/callbacks/manager';
+import { BaseMessage } from '@langchain/core/messages';
+import type { ChatResult } from '@langchain/core/outputs';
+import { AzureChatOpenAI, AzureOpenAI } from '@langchain/openai';
+import { OpenAiChatClient as OpenAiChatClientBase } from '@sap-ai-sdk/foundation-models';
+import { mapLangchainToAiClient, mapResponseToChatResult } from './util.js';
+import type { OpenAiChatModelInput, OpenAiChatCallOptions } from './types.js';
+
+/**
+ * OpenAI Language Model Wrapper to generate texts.
+ */
+export class OpenAiChatClient extends AzureChatOpenAI {
```

Comment:
I agree that it would be fine to switch to ChatOpenAi, the question is what do we prefer to support, users switch from OpenAI or AzureOpenAi, on the implementation side its easy to go either way.

Reply:
Do you have a preference?

Reply:
I would prefer not to offer non-functional API wherever we can 😄

Reply:
Well the don't offer the non-functional part, it's omitted from the type, but if users make the switch, it still works as we overwrite the additional parts.

Reply:
Urgh I see, things like `azureOpenAIBasePath` are in addition also defined on `ChatOpenAI`, not just `AzureChatOpenAI` 🙄

But eitheray, `AzureChatOpenAI` is pretty useless, right? We could even consider to directly use `BaseChatModel` with our own types. I don't see much that `ChatOpenAI` is adding that we can't easily add ourselves based on the already existing client...

Reply:
It's 100% useless and arguably less clean than using BaseChatModel, however, I did it to make the switch as easy as possible.
Do you think going with BaseChatModel is the preferred approach over easier migration?

Reply:
I guess so? I think users just rename `BTPOpenAIChat` to `ChatOpenAI` or whatever we decide on, and all the passed parameters would remain valid, regardless of which approach we take, wright?

Reply:
That is true, however, the default behavior would change.
LangChain's OpenAI client sets lots of defaults (e.g. temperature to 0.7, ...) whereas we would not do that if we just use the base model.

Reply:
Okay, I'd think changing the defaults would be an acceptable change. Maybe we need some research on what reasonable defaults should be anyway and also apply them in the standalone client. For me it would feel unexpected if e.g. the temperature default changes between the standalone client and langchain usage..

Reply:
I would expect the server to have reasonable defaults, as it already allows for these values to be undefined, therefore these defaults seem to be redundant IMO.